core/domain/email_manager.py

```diff
@@ -248,7 +248,7 @@ def log_new_error(*args, **kwargs):
 }
 
 
-def _require_sender_id_is_valid(intent, sender_id):
+def require_sender_id_is_valid(intent, sender_id):
```

Comment:
Also, what was the rationale for simply making this public? We don't want to flip functions so that it is public for the sake of completing the issue.

Reply:
Hi Kevin, 

Since there wasn't an existing public function implementing _require_sender_id_is_valid, our initial plan was to use other public functions that use this function in email_manager.py. We attempted to call one of the two other functions that calls _require_sender_id_is_valid. Neither of those functions are public, but there are public functions that call these functions, so we would use send_user_query_mail to call _send_bulk_mail which is one of the two function that calls _require_sender_id_is_valid. 

This did not work for either of the two functions. We discovered that _require_sender_id_is_valid checks if the sender_id has sufficient permission for a particular intent by consulting the SENDER_VALIDATORS dict. However, there exists no public methods utilizing any intents requiring is_at_least_moderator (EMAIL_INTENT_UNPUBLISH_EXPLORATION and EMAIL_INTENT_DELETE_EXPLORATION). Moreover, all public methods utilizing an intent requiring SYSTEM_COMMITTER_ID do not have a sender_id field we can pass in; these methods already fix SYSTEM_COMMITTER_ID as an argument for sender_id, so we can not pass in an invalid sender_id's without changing their signatures. 

The only public methods passing in arbitrary sender_id or intent into  _require_sender_id_is_valid are send_user_query_email and send_moderator_action_email. However, the bulk email model used in send_user_query_email does not accept the intents we want to test. Also, send_moderator_action_email won't do either since it checks if the intent is in feconf.VALID_MODERATOR_ACTIONS.

Thus, we thought it was best to convert _require_sender_id_is_valid to public after also speaking with @kevintab95 via email about the issue we ran into.

Reply:
Okay, thanks for the explanation. I'll leave the ultimate decision to @aks681 (who is the code owner for this file). My intent was simply to make sure that this was a thought out decision.